hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -866,11 +866,11 @@ public void validateInsertSchema() throws HoodieInsertException {
   }
 
   public HoodieFileFormat getBaseFileFormat() {
-    return metaClient.getTableConfig().getBaseFileFormat();
-  }
-
-  public HoodieFileFormat getLogFileFormat() {
-    return metaClient.getTableConfig().getLogFileFormat();
+    HoodieTableConfig tableConfig = metaClient.getTableConfig();
+    if (tableConfig.contains(HoodieTableConfig.BASE_FILE_FORMAT)) {
+      return metaClient.getTableConfig().getBaseFileFormat();
+    }
+    return config.getBaseFileFormat();
```

Comment:
Should the table config has higher priority or the vise-versa?

Reply:
checking table config first for backward compatibility.

Reply:
write config should always have higher priority, compatibility does not make sense to me.

Reply:
My thinking is as follows: Suppose you've an existing table in ORC, and now you're upgrading to the newer version with this write config. You would not have set the write config and then default value will be set to Parquet, even though you intend to write in ORC. 
But I understand your concern. I think a better way here would be to:
1. Check table config if multi file formats are enabled. If so, then use writeConfig.getBaseFileFormat().
2. Else, use the tableConfig.getBaseFileFormat.
Wdyt?